ddl/partition.go

```diff
@@ -42,6 +42,17 @@ func buildTablePartitionInfo(ctx sessionctx.Context, d *ddl, s *ast.CreateTableS
 	if s.Partition == nil {
 		return nil, nil
 	}
+
+	// force-discard the unsupported types, even when @@tidb_enable_table_partition = 'on'
+	switch s.Partition.Tp {
+	case model.PartitionTypeKey:
+		// can't create a warning for KEY partition, it will fail an integration test :/
```

Comment:
Which test? it asserts the warning to be null?

Reply:
@tiancaiamao `partition_innodb.test` from tidb-test. Maybe we should change the test case instead?